### PR TITLE
Azure Core May GA Release

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
 # Release History
 
+## 1.9.0 (2023-05-04)
+
+### Features Added
+
+- Added the ability to ignore invalid certificate common name for TLS connections in WinHTTP transport.
+- Added `DisableTlsCertificateValidation` in `TransportOptions`.
+- Added `TokenCredential::GetCredentialName()` to be utilized in diagnostic messages. If you have any custom implementations of `TokenCredential`, it is recommended to pass the name of your credential to `TokenCredential` constructor. The old parameterless constructor is deprecated.
+- Added support for challenge-based and multi-tenant authentication.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CHANGELOG.md:10]: I think we should consider keeping this in beta until next month, or at least after an api review is done.

We should close on that discussion.

**Comment by ahsonkhan** [comment]
Why are we GA'ing from a hot-fix branch? We seem to have inverted it. Unlike betas, features, or actual bug/hot fixes, GA releases typically should come out of main, not branches.

**Comment by antkmsft** [comment]
@ahsonkhan, we just did two GA releases from the hotfix branch. This is not technically different. Release is tied to a tag, the branch gets deleted. Even if we hold this for month, in May there could be more Beta commits, so in the end we will end up cherry-picking the current set of commits in `main`.
The same way, it is not possible for us to GA this month from the top of `main`, unless we also include beta features (GUID related).
As of holding this in for another month, this is an important decision which would hold all the releases for May - core 1.9.0, identity, 4 keyvault releases. I do not see why we should hold GA - we have released it as Beta in April. It has been baking for a month. This has been the usual cadence. We could've held it for some time if we knew an even that we are waiting for, but what are we waiting to happen in this month?
As of API review - we will of course review the API this month before GA'ing.

**Comment by antkmsft** [comment]
@ahsonkhan, we have API reviewed and approved -

Core: https://apiview.dev/Assemblies/Review/e70fdd6e25954ba8b474f7e7b0fc022b?revisionId=979381c7dd384dcbb60a78fc0ae7da50&doc=False

Identity: https://apiview.dev/Assemblies/Review/89dc2c0761c34a50a6156e2198b78f1b?revisionId=a1516db7809049e0a4e60dee9a38b603&doc=False

File: sdk/core/azure-core/CHANGELOG.md (hunk 1 above)
Comment (antkmsft) [reply sdk/core/azure-core/CHANGELOG.md:10]: I replied in two comments below.

**Comment by ahsonkhan** [comment]
The "hotfix" release approach is a very specific corner case, particularly for out-of-band releases with a targeted fix. This approach shouldn't be the de-facto pattern for any subsequent GA releases. We are subtly, but surely, venturing off to a different release strategy than what has been done previously (and in other languages).

At the very least, the branch name we are creating should be changed to remove "hotfix".

See this SDK-wide guideline
https://github.com/Azure/azure-sdk/blob/cc4c73c0965131f0a68306da1987bda57ea4caae/docs/policies/repobranching.md#release-branches

Hotfix and Release branches are similar, but not the same in when they are used. We should treat these differently.

With the piecemeal approach we are trying here, we have inverted the pattern of what ships (compared to what is being tested regularly), and we should come up with a coherent strategy for what is going to GA vs beta, and when we want to deviate from the guideline. For example, this bug fix belongs in a GA release rather than pushed off into next months beta (and that likely applies to most bug fixes or non-feature/new api surface commits to main):
https://github.com/Azure/azure-sdk-for-cpp/pull/4577/files#diff-cc1b46c667702f97b0f4104c7637ccca9da686d2df0cd9e47a39f12ced659897L14-R12

**Comment by antkmsft** [comment]
@ahsonkhan,

1. As of the Uuid fix - I shared proposed changelogs over email on Monday, and this is the first time that I am hearing that we need this fix for 1.9.0. It can be done - I'll cherry-pick it for the 1.9.0 GA PR, and will update changelog for the beta release PR to have it chronologically under 1.9.0.

2. `hotfix/azure-core_1.9.0` branch can be renamed to `release/azure-core_1.9.0` right from the GitHub UI, and it will automatically update the branch that this PR is targeting, I can do that. The branch will be deleted after the release anyways.

3. I talked to Ben, and this cherry-picking approach for the releases is the one that he described to us. Other approach he mentioned was doing all development in a separate branch, and cherry-picking commits from there. It is more complicated logistically, it means that we have to cherry-pick for every release even if we are doing just one release as before, it means that customer no longer gets the latest features as they clone our repo.
Cherry-picking for the release is pretty straighforward - we have done it two times already, and we figured out what to cherry-pick and how to get good diffs for the release PRs. I think it is reasonably the best approach, it does what we need in a reasonable way. It will create a non-problematic release in the end.
Besides that, what alternative do you have in mind? We need to cut a GA release without some features that chronologically sit between the commits that we want to release, here comes the idea of a release branch and cherry-picking commits for it. Our `main` is still the bleeding edge - for our development and for our customers. Therefore, cherry-picking for the release branch. Cherry-picking is a tool that gives us the reliable results. If you have feedback that some of the commits that should be here were not picked, that's one thing, and this is why I have this PR. But I don't agree with you that we somehow can't trust cherry-picking in general and that cherry-picking produces somehow inferior builds. If we did not miss a thing that we wanted to be in the release, it is just as good as if chronologically we only merged the release-worthy commits to `main` and released from there.

**Comment by ahsonkhan** [comment]
> I'll cherry-pick it for the 1.9.0 GA PR
> branch can be renamed

Thanks!

> I talked to Ben, and this cherry-picking approach for the releases is the one that he described to us.
> But I don't agree with you that we somehow can't trust cherry-picking in general and that cherry-picking produces somehow inferior builds. 

I see why the cherry-picking approach works well here. I would love to talk about this a bit more, offline, to understand the plan moving forward for cherry-picking. It's likely my concern here is already addressed :)

**Pull request merged**